astroquery/vizier/core.py

```diff
@@ -345,6 +345,9 @@ def get_catalog_metadata(self, *, catalog=None, get_query_payload=False):
         if get_query_payload:
             return metadata
         result = metadata.execute().to_table()
+        if len(result) == 0:
+            raise ValueError(f"'{catalog}' was not found in VizieR. Valid catalog names "
```

Comment:
We do have both `NoResultsWarning` as well as `EmptyResponseError`. 

What do you think using either of those? Given that here an empty response is very likely due to a user typo rather than not having coverage (as for most of the use cases for the Warning), I would be OK with raising the exception rather than just a warning.

Reply:
Ah thanks, I also found `ValueError` quite vague :)
I chose `EmptyResponseError` because catalogs always have metadata, so there is no way this would be just a lack of coverage.